variables.tf

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "project_id" {
+  description = "The project ID to host the cluster in (required)"
+}
+
+variable "name" {
+  description = "The name of the cluster (required)"
+}
+
+variable "description" {
+  description = "The description of the cluster"
+  default     = ""
+}
+
+variable "regional" {
+  description = "Whether is a regional cluster (zonal cluster if set false. WARNING: changing this after cluster creation is destructive!)"
+  default     = true
+}
+
+variable "region" {
+  description = "The region to host the cluster in (required)"
+}
+
+variable "zones" {
+  type        = "list"
+  description = "The zones to host the cluster in (optional if regional cluster / required if zonal)"
+  default     = []
+}
+
+variable "network" {
+  description = "The VPC network to host the cluster in (required)"
+}
+
+variable "network_project_id" {
+  description = "The project ID of the shared VPC's host (for shared vpc support)"
+  default     = ""
+}
+
+variable "subnetwork" {
+  description = "The subnetwork to host the cluster in (required)"
+}
+
+variable "kubernetes_version" {
+  description = "The Kubernetes version of the masters. If set to 'latest' it will pull latest available version in the selected region."
+  default     = "1.10.6-gke.2"
```

Comment:
@ryanckoch Should this default to "latest" - we probably don't want to have to constantly update this default.